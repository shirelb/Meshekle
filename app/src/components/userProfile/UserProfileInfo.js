import React, {Component} from 'react';
import {Image, Linking, Modal, ScrollView, StyleSheet, TextInput, TouchableOpacity, View} from "react-native";
import {Icon, Text} from "react-native-elements";
import moment from "moment";
import mappers from "../../shared/mappers";


export default class UserProfileInfo extends Component {
    constructor(props) {
        super(props);

        this.state = {
            modalVisible: this.props.modalVisible,
            user: this.props.user,
            roles: [],
            openedFrom: this.props.openedFrom,
        }
    }

    componentWillReceiveProps(nextProps, nextContext) {
        let roles = [];
        nextProps.user.ServiceProviders ?
            nextProps.user.ServiceProviders.forEach(provider => {
                roles.push(mappers.serviceProviderRolesMapper(provider.role));
            }) : null;
        this.setState({
            user: nextProps.user,
            roles: roles,
        });

        this.setState({
            modalVisible: this.props.modalVisible,
            openedFrom: this.props.openedFrom,
        });
    }

    setModalVisible(visible) {
        this.setState({modalVisible: visible});
    }


    render() {
        const user = this.state.user;
        // console.log('user ', user);

        return (
            <Modal
                animationType="slide"
                transparent={false}
                visible={this.state.modalVisible}
                onRequestClose={() => {
                    this.setModalVisible(!this.state.modalVisible);
                    this.setState({user: {}})
                }}
            >
                <View style={{marginTop: 20}}>
                    <ScrollView>
                        <View style={{flex: 1, flexDirection: 'row', justifyContent: 'space-between',}}>
                            <View style={{marginLeft: 20, marginTop: 20}}>
                                <TouchableOpacity
                                    onPress={() => {
                                        this.setModalVisible(!this.state.modalVisible);
                                        this.setState({user: {}})
                                    }}
                                >
                                    <Icon
                                        name='arrow-forward'
                                        size={50}
                                    />
                                </TouchableOpacity>
                            </View>
                            <View style={styles.circle1}>
                                <View style={styles.circle2}>
                                    <View style={styles.circle3}>
                                        <Image style={styles.avatar}
                                               source={{uri: user.image}}
                                        />
                                    </View>
                                </View>
                            </View>
                        </View>


                        <View style={{flex: 1, flexDirection: 'row', justifyContent: 'space-around',}}>
                            <Text h3 style={styles.textName}>{user.fullname}</Text>

                            {this.state.openedFrom === "DrawerMenu" ?
                                <TouchableOpacity
                                    onPress={() => {
                                        // this.setModalVisible(!this.state.modalVisible);
                                        this.setState({modalVisible: false, user: {}});
                                        this.props.setFormModalVisible();
                                    }}
                                    style={{marginTop: 10}}
                                >
                                    <Icon
                                        name='edit'
                                        size={30}
                                    />
                                </TouchableOpacity>
                                : <View style={{width: 50}}/>
                            }
                        </View>

                        <View
                            style={{
                                height: 2,
                                width: "86%",
                                backgroundColor: "#CED0CE",
                                marginLeft: "14%"
                            }}
                        />

                        {this.state.openedFrom === "DrawerMenu" ?
                            <View style={styles.viewInfo}>
                                <Icon name='lock' style={styles.iconInfo}/>
                                <TextInput
                                    editable={false}
                                    secureTextEntry
                                    style={styles.textInput}
                                    underlineColorAndroid="transparent"
                                >
                                    {user.password}
                                </TextInput>
                            </View> : null
                        }
                        <View pointerEvents={this.state.openedFrom === "DrawerMenu" ? 'none' : 'auto'}>
                            <TouchableOpacity style={styles.viewInfo}
                                              onPress={() => Linking.openURL(`tel:${user.cellphone}`)}>
                                <Icon name='phone-android' style={styles.iconInfo}/>
                                <Text style={styles.textInfo}>
                                    {user.cellphone}
                                </Text>
                            </TouchableOpacity>
                        </View>
                        <View pointerEvents={this.state.openedFrom === "DrawerMenu" ? 'none' : 'auto'}>
                            <TouchableOpacity style={styles.viewInfo}
                                              onPress={() => Linking.openURL(`tel:${user.phone}`)}>
                                <Icon name='phone' style={styles.iconInfo}/>
                                <Text style={styles.textInfo}>{user.phone}</Text>
                            </TouchableOpacity>
                        </View>
                        <View pointerEvents={this.state.openedFrom === "DrawerMenu" ? 'none' : 'auto'}>
                            <TouchableOpacity style={styles.viewInfo}
                                              onPress={() => Linking.openURL(`mailto:${user.email}`)}>
                                {/*title="support@example.com">*/}
                                <Icon name='mail-outline' style={styles.iconInfo}/>
                                <Text style={styles.textInfo}>{user.email}</Text>
                            </TouchableOpacity>
                        </View>
                        <View style={styles.viewInfo}>
                            <Icon name='envelope-square' type='font-awesome'
                                  style={styles.iconInfo}/>
                            <Text style={styles.textInfo}>{user.mailbox}</Text>
                        </View>
                        <View style={styles.viewInfo}>
                            <Icon name='cake' style={styles.iconInfo}/>
                            <Text style={styles.textInfo}>{moment(user.bornDate).format('DD/MM')}</Text>
                        </View>

                        {this.state.roles.length > 0 ?
                            <View style={styles.viewInfo}>
                                <Icon name='work' style={styles.iconInfo}/>
                                <Text style={styles.textInfo}>
                                    {this.state.roles.join(", ")}
                                </Text>
                            </View> : null
                        }

                        {/*<View style={{marginTop: 20}}>
                            <Button
                                label='חזור'
                                onPress={() => {
                                    this.setModalVisible(!this.state.modalVisible);
                                                        this.setState({user: {}})
                                }}
                            />
                        </View>*/}

                    </ScrollView>

                </View>
            </Modal>
        );
    }
}

const styles = StyleSheet.create({
    textTitle: {
        textAlign: 'center',
        color: '#050505',
        fontWeight: 'bold',
    },


    circle1: {
        width: 250,
        height: 250,
        borderRadius: 250 / 2,
        backgroundColor: '#2a674f',
        alignSelf: 'flex-end',
        marginTop: -50,
        marginRight: -50,
        position: 'relative',
    },
    circle2: {
        width: 240,
        height: 240,
        borderRadius: 240 / 2,
        backgroundColor: '#388b6b',
        alignSelf: 'flex-end',
        marginTop: -10,
        marginRight: -10,
        position: 'absolute',
    },
    circle3: {
        width: 225,
        height: 225,
        borderRadius: 225 / 2,
        backgroundColor: '#4aba91',
        alignSelf: 'flex-end',
        marginTop: 0,
        marginRight: 0,
        position: 'absolute',
    },

    avatar: {
        width: 150,
        height: 150,
        borderRadius: 150 / 2,
        borderWidth: 4,
        borderColor: "white",
        position: 'absolute',
        marginTop: 60,
        marginRight: 40,
        marginLeft: 20,
    },

    viewInfo: {
        // width: 25,
        paddingVertical: 10,
        paddingHorizontal: 15,
        flexDirection: "row",
        alignItems: "center"
    },

    iconInfo: {
        fontSize: 21,
        // color: '#bbbbbb',
        textAlign: 'center',
        marginLeft: 20,
        marginRight: 20,
    },

    textInfo: {
        // color: '#bbbbbb',
        fontSize: 18,
        textAlign: 'center',
        marginLeft: 20,
        marginRight: 20,
    },

    textName: {
        marginLeft: 20,
        marginRight: 20,
    },

    textInput: {
        height: 50,
        // borderRadius: 10 ,
        // borderWidth: 2,
        // borderColor: '#009688',
        backgroundColor: "#FFF",
        // marginBottom: 10,
        color: '#838383',
        fontSize: 18,
        textAlign: 'center',
        marginLeft: 20,
        marginRight: 20,
    }

});
