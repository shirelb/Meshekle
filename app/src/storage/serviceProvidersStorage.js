import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getServiceProviders = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/serviceProviders`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('load appointments error ', error)
        });
};


export default {getServiceProviders};
