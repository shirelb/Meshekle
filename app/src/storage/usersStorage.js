import axios from "axios";
import {SERVER_URL} from "../shared/constants";

var getUserEvents = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/events/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user events error ', error)
        });
};

var getUserById = function (userId, userHeaders) {
    return axios.get(`${SERVER_URL}/api/users/userId/${userId}`,
        {headers: userHeaders}
    )
        .then(response => {
            return response;
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var getUsers = function (userHeaders) {
    return axios.get(`${SERVER_URL}/api/users`,
        {headers: userHeaders}
    )
        .then(response => {
            return response.data.filter(user => user.userId !== 1 && user.userId !== "1");
        })
        .catch(error => {
            console.log('get user by id error ', error)
        });
};

var userValidToken = function (token) {
    return axios.post(`${SERVER_URL}/api/users/validToken`,
        {
            "token": token,
        },
    );
}

var userLogin = function (userId, password) {
    return axios.post(`${SERVER_URL}/api/users/login/authenticate`,
        {
            "userId": userId,
            "password": password
        },
    );
}


export default {getUserEvents, getUserById, userValidToken, userLogin, getUsers};
