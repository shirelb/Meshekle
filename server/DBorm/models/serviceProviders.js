module.exports = (sequelize, type) => {
    return sequelize.define('ServiceProviders', {
        serviceProviderId:{
            type: type.INTEGER,
            primaryKey: true,
            autoIncrement: true,
            allowNull: false,
            required: true
        },
        role : {
            type: type.STRING,
            allowNull: false,
            required: true
        },
        userId: {
            type: type.INTEGER,
            allowNull: false,
            required: true
        },
        operationTime: {
            type: type.STRING,
            allowNull: false,
            required: true
        },
        phoneNumber: {
            type: type.STRING,
            allowNull: false,
            required: true
        },
    });
};