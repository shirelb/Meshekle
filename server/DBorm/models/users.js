module.exports = (sequelize, type) => {
    return sequelize.define('Users', {
        userId:{
            type: type.STRING,
            primaryKey: true,
            required: true
        },
        fullname: {
            type: type.STRING,
            required: true
        },
        password: {
            type: type.STRING,
            required: true
        },
        email: {
            type: type.STRING,
            required: true
        },
        mailbox: {
            type: type.INTEGER,
            required: true
        },
        cellphone: {
            type: type.STRING,
            required: true
        },
        phone: {
            type: type.STRING
        },
        bornDate: {
            type: type.DATE,
            required: true
        },
        active: {
            type: type.BOOLEAN,
            required: true,
            defaultValue: true
        },
        image: {
            type: type.BLOB,
        }
    });
};