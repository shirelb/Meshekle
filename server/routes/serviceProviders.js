var express = require('express');
var router = express.Router();
const Sequelize = require('sequelize');
const Op = Sequelize.Op;
const {Users, ServiceProviders, AppointmentRequests, Appointments, Incidents, Events} = require('../DBorm/DBorm');

/* GET users listing. */
router.get('/', function(req, res, next) {
  res.send('respond with a resource serviceProviders');
});


/* POST user . */
router.post('/users/add', function (req, res, next) {
    Users.create({
        userId: req.body.userId,
        fullname: req.body.fullname,
        password: req.body.password,
        email: req.body.email,
        mailbox: req.body.mailbox,
        cellphone: req.body.cellphone,
        phone: req.body.phone,
        bornDate: new Date(req.body.bornDate),
        active: true,
    })
        .then((newUser) => {
            console.log('New user' + newUser.username + ', with id ' + newUser.userId + ' has been created.');
            res.status(200).send({"message": "User successfully added!", newUser});
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

/* POST user . */
router.post('/add', function (req, res, next) {
    ServiceProviders.create({
        serviceProviderId: req.body.serviceProviderId,
        userId: req.body.userId,
        role: req.body.role,
        operationTime: req.body.operationTime,
        phoneNumber: req.body.phoneNumber,
        appointmentWayType: req.body.appointmentWayType,
        active: true,
    })
        .then((newUser) => {
            console.log('New user' + newUser.username + ', with id ' + newUser.userId + ' has been created.');
            res.status(200).send({"message": "User successfully added!", newUser});
        })
        .catch(err => {
            console.log(err);
            res.status(500).send(err);
        })
});

module.exports = router;
