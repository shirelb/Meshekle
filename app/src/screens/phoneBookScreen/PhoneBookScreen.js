import React, {Component} from 'react';
import {FlatList, StyleSheet, View} from 'react-native';
import {Icon, List, ListItem, SearchBar} from 'react-native-elements';
import phoneStorage from "react-native-simple-store";
import UserProfileInfo from "../../components/userProfile/UserProfileInfo";
import usersStorage from "../../storage/usersStorage";
import serviceProvidersStorage from "../../storage/serviceProvidersStorage";
import mappers from "../../shared/mappers";
import {APP_SOCKET} from "../../shared/constants";


export default class PhoneBookScreen extends Component {
    constructor(props) {
        super(props);

        this.state = {
            users: [],
            serviceProviders: [],
            infoModal: false,
            userSelected: {},
            noUserFound: false,
        };

        this.users = [];
        this.serviceProviders = [];
        this.openUserInfo = this.openUserInfo.bind(this);
    }

    componentDidMount() {
        phoneStorage.get('userData')
            .then(userData => {
                // console.log('agenda componentDidMount userData ', userData);
                this.userHeaders = {
                    'Authorization': 'Bearer ' + userData.token
                };
                this.userId = userData.userId;
                this.loadUsers();
            });

        APP_SOCKET.on("getServiceProviders", this.loadUsers.bind(this));
        APP_SOCKET.on("getUsers", this.loadUsers.bind(this));
    }

    componentWillUnmount() {
        APP_SOCKET.off("getServiceProviders");
        APP_SOCKET.off("getUsers");
    }

    loadUsers() {
        usersStorage.getUsers(this.userHeaders)
            .then(users => {
                console.log("phonebook users ", users);

                users.forEach(user => {
                    if (user.ServiceProviders.length > 0) {
                        let roles = [];
                        user.ServiceProviders.forEach(provider => {
                            if (provider.active)
                                roles.push(mappers.serviceProviderRolesMapper(provider.role));
                        });
                        user.roles = roles;
                    }
                });

                this.setState({
                    users: users.filter(user => user.active === true).sort((a, b) => a.fullname !== b.fullname ? a.fullname < b.fullname ? -1 : 1 : 0),
                });

                this.users = users;

            })
    };

    loadServiceProviders() {
        serviceProvidersStorage.getServiceProviders(this.userHeaders)
            .then(serviceProviders => {
                // let serviceProviders = response.data;

                this.setState({
                    serviceProviders: serviceProviders,
                });

                this.serviceProviders = serviceProviders;
            })
    };

    openUserInfo = (user) => {
        this.setState({
            infoModal: true,
            userSelected: user
        });
    };

    renderSeparator = () => {
        return (
            <View
                style={{
                    height: 1,
                    width: "86%",
                    backgroundColor: "#CED0CE",
                    marginLeft: "14%"
                }}
            />
        );
    };

    updateSearch = search => {
        // console.log("in search ", search);
        this.setState({infoModal: false});
        let users = this.state.users;
        let filteredByNameOrRole = users.filter((item) => {
            let roleFound = false;
            item.ServiceProviders.forEach(provider => {
                if (mappers.serviceProviderRolesMapper(provider.role).includes(search)) {
                    roleFound = true;
                }
            });
            return item.fullname.includes(search) || roleFound;
        });
        if (!search || search === '') {
            this.setState({
                users: this.users
            })
        } else if (!Array.isArray(filteredByNameOrRole) && !filteredByNameOrRole.length) {
            // set no data flag to true so as to render flatlist conditionally
            this.setState({
                noUserFound: true
            })
        } else if (Array.isArray(filteredByNameOrRole)) {
            this.setState({
                noUserFound: false,
                users: filteredByNameOrRole
            })
        }
    };

    renderHeader = () => {
        return <SearchBar
            placeholder="חפש..."
            lightTheme
            onChangeText={this.updateSearch.bind(this)}
            // round
        />;
    };

    renderRow = ({item}) => {
        console.log("item ", item);

        if (item.image !== null)
            return (
                <ListItem
                    roundAvatar
                    title={item.fullname}
                    subtitle={item.roles ? item.roles.join(", ") : ""}
                    avatar={{uri: item.image}}
                    onPress={() => this.openUserInfo(item)}
                    containerStyle={{borderBottomWidth: 0}}
                    rightIcon={<Icon name={'chevron-left'}/>}
                />
            );
        else
            return (
                <ListItem
                    roundAvatar
                    title={item.fullname}
                    subtitle={item.roles ? item.roles.join(", ") : ""}
                    avatar={{uri: "https://user-images.githubusercontent.com/30195/34457818-8f7d8c76-ed82-11e7-8474-3825118a776d.png"}}
                    onPress={() => this.openUserInfo(item)}
                    containerStyle={{borderBottomWidth: 0}}
                    rightIcon={<Icon name={'chevron-left'}/>}
                />
            )
    };

    render() {
        return (
            <View>
                <List containerStyle={{borderTopWidth: 0, borderBottomWidth: 0}}>
                    {this.state.noUserFound ? <Text>לא נמצאו תוצאות</Text> :
                        <FlatList
                            data={this.state.users}
                            renderItem={this.renderRow}
                            keyExtractor={item => item.userId}
                            ItemSeparatorComponent={this.renderSeparator}
                            ListHeaderComponent={this.renderHeader}
                        />
                    }
                </List>
                <UserProfileInfo
                    modalVisible={this.state.infoModal}
                    user={this.state.userSelected}
                    openedFrom={"PhoneBookScreen"}
                />
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        flex: 1,
        justifyContent: 'center',
        alignItems: 'center'
    }
});

