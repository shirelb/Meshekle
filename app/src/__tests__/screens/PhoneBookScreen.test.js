import React from 'react';

import {FlatList} from 'react-native';
import {List} from 'react-native-elements';

import {shallow} from "enzyme/build";
import serviceProvidersStorage from "../../storage/serviceProvidersStorage";
import usersStorage from "../../storage/usersStorage";
import users from "../jsons/users";
import serviceProviders from "../jsons/serviceProviders";
import PhoneBookScreen from "../../screens/phoneBookScreen/PhoneBookScreen";
import phoneStorage from "react-native-simple-store";
import UserProfileInfo from "../../components/userProfile/UserProfileInfo";

jest.mock("react-native-simple-store");
jest.mock("../../storage/usersStorage");
jest.mock("../../storage/serviceProvidersStorage");

const flushPromises = () => new Promise(setImmediate);

describe("PhoneBookScreen should", () => {
    let wrapper = null;
    let componentInstance = null;
    const props = {};
    const userTest = users[41];
    const mockStore = {
        userData: {
            serviceProviderId: "549963652",
            userId: "549963652",
            token: "some token"
        }
    };

    phoneStorage.get = jest.fn().mockImplementation((key) => Promise.resolve(mockStore[key]));
    usersStorage.getUsers = jest.fn().mockImplementation(()=>Promise.resolve(users));
    usersStorage.getUserByUserID = jest.fn().mockImplementation((userId) => Promise.resolve(users.filter(user => user.userId === userId)[0]));
    serviceProvidersStorage.getServiceProviders = jest.fn().mockResolvedValue(serviceProviders);
    serviceProvidersStorage.getServiceProviderUserDetails = jest.fn().mockImplementation((serviceProviderId) => Promise.resolve({data: users.filter(user => user.userId === serviceProviderId)[0]}));
    serviceProvidersStorage.getServiceProviderById = jest.fn().mockImplementation((serviceProviderId) => Promise.resolve(serviceProviders.filter(provider =>
        provider.serviceProviderId === serviceProviderId)));

    beforeAll(() => {
    });

    afterAll(() => {
    });

    beforeEach(() => {
        jest.clearAllMocks();
    });

    afterEach(() => {
    });

    it('match snapshot', async () => {
        wrapper = shallow(<PhoneBookScreen/>);
        expect(wrapper).toMatchSnapshot();
    });

    it("mounted with the right data", async () => {
        wrapper = await shallow(<PhoneBookScreen/>);
        componentInstance = wrapper.instance();

        await wrapper.update();

        expect(componentInstance.state.users.length).toEqual(513);
        expect(componentInstance.state.infoModal).toEqual(false);
        expect(componentInstance.state.userSelected).toEqual({});
        expect(componentInstance.state.noUserFound).toEqual(false);
    });

    it("render with what the user see", async () => {
        wrapper = shallow(<PhoneBookScreen/>);
        componentInstance = wrapper.instance();

        expect(wrapper.find(List)).toHaveLength(1);
        expect(wrapper.find(UserProfileInfo)).toHaveLength(1);
        expect(wrapper.find(FlatList)).toHaveLength(1);
    });

    it("search for users with name that has ra", async () => {
        wrapper = await shallow(<PhoneBookScreen/>);
        componentInstance = wrapper.instance();

        await wrapper.update();
        const updateSearchSpy = jest.spyOn(componentInstance, 'updateSearch');

        await componentInstance.updateSearch('ra');

        expect(updateSearchSpy).toHaveBeenCalled();
        expect(componentInstance.state.users.length).toEqual(39);
    });

    it("search for users with role מספרה", async () => {
        wrapper = await shallow(<PhoneBookScreen/>);
        componentInstance = wrapper.instance();

        await wrapper.update();

        const updateSearchSpy = jest.spyOn(componentInstance, 'updateSearch');

        await componentInstance.updateSearch('מספרה');

        expect(updateSearchSpy).toHaveBeenCalled();
        expect(componentInstance.state.users.length).toEqual(45);
    });

    it("open UserProfileInfo on click user", async () => {
        wrapper = shallow(<PhoneBookScreen/>);
        componentInstance = wrapper.instance();

        const openUserInfoSpy = jest.spyOn(componentInstance, 'openUserInfo');

        componentInstance.openUserInfo(userTest);

        expect(openUserInfoSpy).toHaveBeenCalled();
        expect(openUserInfoSpy).toHaveBeenCalledWith(userTest);
    });

});
