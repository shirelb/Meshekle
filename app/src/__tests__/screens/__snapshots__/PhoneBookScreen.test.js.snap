// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PhoneBookScreen should match snapshot 1`] = `
<View>
  <List
    containerStyle={
      Object {
        "borderBottomWidth": 0,
        "borderTopWidth": 0,
      }
    }
  >
    <FlatList
      ItemSeparatorComponent={[Function]}
      ListHeaderComponent={[Function]}
      data={Array []}
      disableVirtualization={false}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      numColumns={1}
      onEndReachedThreshold={2}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      updateCellsBatchingPeriod={50}
      windowSize={21}
    />
  </List>
  <UserProfileInfo
    modalVisible={false}
    openedFrom="PhoneBookScreen"
    user={Object {}}
  />
</View>
`;
